image_search_engine/src/data/loader.py

```diff
@@ -0,0 +1,53 @@
+import torch
+import cv2
+import albumentations as A
+from image_search_engine.src import config as CFG
+
+class CLIPLoader(torch.utils.data.Dataset):
+    def __init__(self, image_filenames, captions, tokenizer, transforms):
+        """
+        image_filenames and cpations must have the same length; so, if there are
+        multiple captions for each image, the image_filenames must have repetitive
+        file names
+        """
+
+        self.image_filenames = image_filenames
+        self.captions = list(captions)
+        self.encoded_captions = tokenizer(
+            list(captions), padding=True, truncation=True, max_length=CFG.MAX_LENGTH
+        )
+        self.transforms = transforms
+
+    def __getitem__(self, idx):
+        item = {
+            key: torch.tensor(values[idx])
+            for key, values in self.encoded_captions.items()
+        }
+
+        image = cv2.imread(f"{CFG.RAW_PATH}/{self.image_filenames[idx]}")
+        image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+        image = self.transforms(image=image)["image"]
+        item["image"] = torch.tensor(image).permute(2, 0, 1).float()
+        item["caption"] = self.captions[idx]
+
+        return item
+
+    def __len__(self):
+        return len(self.captions)
+
+
+def get_transforms(mode="train"):
+    if mode == "train":
+        return A.Compose(
+            [
+                A.Resize(CFG.IMAGE_SIZE, CFG.IMAGE_SIZE, always_apply=True),
```

Comment:
I would image_size as a parameter and pass CFG.IMAGE_SIZE as an argument when calling `get_transforms`.